src/AutoRest.CSharp/Common/Generation/Types/CSharpType.cs

```diff
@@ -102,7 +102,7 @@ public CSharpType(TypeProvider implementation, string ns, string name, bool isVa
         public bool IsLiteral { get; init; }
         public Constant? Literal { get; init; }
         public bool IsUnion { get; }
-        public CSharpType[] UnionItemTypes { get; } = Array.Empty<CSharpType>();
+        public IReadOnlyList<CSharpType> UnionItemTypes { get; } = Array.Empty<CSharpType>();
```

Comment:
We have mutable `CSharpType[] Arguments` as a relic of the past when arrays have been used everywhere. Now we use read-only types for public properties that shouldn't be changed. Ideally, `Arguments` should be `IReadOnlyList<CSharpType>` as well, cause changing the list of generic types would cause an error

Reply:
Can we add a work item to fix `Arguments` in that case we can tackle in a separate PR.

Reply:
https://github.com/Azure/autorest.csharp/issues/3958